nix/test-environments/test-script.py

```diff
@@ -0,0 +1,300 @@
+import re
+import shlex
+import json
+import time
+from dataclasses import dataclass
+from typing import List
+from test_driver.errors import RequestedAssertionFailed # type: ignore
+
+# --- Configuration ---
+@dataclass
+class TestConfig:
+    """Centralized configuration for the integration test."""
+    BASE_URL: str = "http://127.0.0.1"
+    INK_PORT: int = 8547
+    SEPOLIA_PORT: int = 8546
+    SEQUENCER_PORT: int = 5553
+
+    # Contract Addresses
+    UPGRADEABLE_PROXY_CONTRACT: str = "0xee5a4826068c5326a7f06fd6c7cbf816f096846c"
+    CHAINLINK_PROXY_CONTRACT: str = "0x9fAb38E38d526c6ba82879c6bDe1c4Fd73378f17"
+    UPGRADEABLE_PROXY_ADFS_CONTRACT: str = "0xADF5aad6faA8f2bFD388D38434Fc45625Ebd9d3b"
+    CL_AGGREGATOR_ADAPTER_CONTRACT: str = "0xcBD6FC059bEDc859d43994F5C221d96F9eD5340f"
+
+    # Service Names
+    REPORTER_SERVICE: str = "blocksense-reporter-a.service"
+    ANVIL_SEPOLIA_SERVICE: str = "blocksense-anvil-ethereum-sepolia.service"
+    ANVIL_INK_SERVICE: str = "blocksense-anvil-ink-sepolia.service"
+    SEQUENCER_SERVICE: str = "blocksense-sequencer.service"
```

Comment:
Might want to make this more dynamic (at least the `-a` part), currently hard-dependening on the nix config outside:

https://github.com/blocksense-network/blocksense/pull/1279/files#diff-f195c6435e680fa841aab93a92956dc6cbc0dce5f9ff63ece2c68e31979ffbc0R21